recipes/opentdf-client/all/conanfile.py

```diff
@@ -36,63 +39,105 @@ def _minimum_cpp_standard(self):
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support."
+                                                 "The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
 
     def requirements(self):
-        self.requires("openssl/1.1.1l@")
-        self.requires("boost/1.76.0@")
-        self.requires("zlib/1.2.11@")
-        self.requires("ms-gsl/2.1.0@")
-        self.requires("libxml2/2.9.10@")
-        self.requires("libarchive/3.5.1@")
-        self.requires("nlohmann_json/3.10.4@")
-        self.requires("jwt-cpp/0.4.0@")
+        self.requires("openssl/1.1.1q")
+        self.requires("boost/1.79.0")
+        self.requires("ms-gsl/2.1.0")
+        self.requires("libxml2/2.9.14")
+        self.requires("nlohmann_json/3.11.1")
+        self.requires("jwt-cpp/0.4.0")
+        if self.options.get_safe("with_libarchive"):
+            self.requires("libarchive/3.6.1")
+        # We do not require libiconv but conan-center only allows 'stock' references, and boost+libxml2
+        # specify differerent versions, which causes a build fail due to the dependency conflict.
+        # Overriding the version here allows a clean build with the stock build settings.
+        # Preferred solution is to set with_libiconv to False
+        if self.options.get_safe("with_libiconv"):
+            self.requires("libiconv/1.17@")
+        # Same for zlib
+        # Preferred solution is to set with_zlib to False
+        if self.options.get_safe("with_zlib"):
+            self.requires("zlib/1.2.12@")
 
     def config_options(self):
+        # Do not require libarchive as of 1.1.0
+        if Version(self.version) < "1.1.0":
+            self.options.with_libarchive = True
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self.options.with_zlib:
+            self.options["libxml2"].zlib = False
+        if not self.options.with_libiconv:
+            self.options["boost"].without_locale = True
+            self.options["boost"].without_log = True
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        # If branch_version is set to True, the supplied version will be treated as a branch name for git clone
+        if self.options.get_safe("branch_version"):
+            self.output.warn("Building branch_version = {}".format(self.version))
+            self.run("git clone git@github.com:opentdf/client-cpp.git --depth 1 --branch " + self.version + " " + self._source_subfolder)
+        else:
+            get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

Comment:
```suggestion
        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```
Use only the regular download, keep the same pattern as used over all recipes.

Reply:
See other comment.  This was to facilitate development against pre-release branches.  The default is still the conandata.yml method.

Please?

Reply:
It's not possible, because CCI don't get local development as part of process. Anyway, I'll talk to Conan Team next, maybe we can find a solution.

Reply:
It is so that I can do that on my local machine, not at CCI.  I suppose I can keep a different version of the recipe in the client-cpp repo and have local builds based on that

Reply:
Updated